sentinel-extension/sentinel-datasource-zookeeper/src/main/java/com/alibaba/csp/sentinel/datasource/zookeeper/ZookeeperDataSource.java

```diff
@@ -165,4 +189,40 @@ public void close() throws Exception {
     private String getPath(String groupId, String dataId) {
         return String.format("/%s/%s", groupId, dataId);
     }
+
+    private String getZkKey(final String serverAddr, final List<AuthInfo> authInfos) {
+        if (authInfos == null || authInfos.size() == 0) {
+            return serverAddr;
+        }
+        Collections.sort(authInfos, new Comparator<AuthInfo>() {
+            @Override
+            public int compare(AuthInfo o1, AuthInfo o2) {
+                int c = o1.getScheme().compareTo(o2.getScheme());
+                if (c != 0) {
+                    return c;
+                }
+                byte[] auth1 = o1.getAuth();
+                byte[] auth2 = o2.getAuth();
+                if (auth1 == null && auth2 == null) {
+                    return 0;
+                }
+                if (auth1 == null) {
+                    return -1;
+                } else if (auth2 == null) {
+                    return 1;
+                }
+                int min = Math.min(auth1.length, auth2.length);
+                for (int i = 0; i < min; i++) {
+                    if (auth1[i] != auth2[i]) {
+                        return auth1[i] - auth2[i];
+                    }
+                }
+                return auth1.length - auth2.length;
+            }
+        });
+
+        return serverAddr + "|" + JSON.toJSONString(authInfos);
```

Comment:
Data source extension modules should not include any JSON libs, which might bring conflict.

Reply:
> Friendly ping :)
Just saw the information, do you mean manual jsonization?

Reply:
We could just use simple logic to generate the unique key.

Reply:
the latest code had committed, please code view  :)  @sczyh30

Reply:
friendly ping :)